src/v/cluster/topics_frontend.cc

```diff
@@ -62,6 +62,26 @@
 #include <sstream>
 #include <system_error>
 
+namespace {
+
+std::optional<std::string_view> get_enterprise_feature(
+  const cluster::custom_assignable_topic_configuration& cfg) {
+    if (
+      cfg.cfg.properties.shadow_indexing
+      != model::shadow_indexing_mode::disabled) {
+        return "tiered storage";
+    }
+    if (cfg.is_recovery_enabled()) {
+        return "topic recovery";
+    }
+    if (cfg.is_read_replica()) {
+        return "remote read replicas";
+    }
```

Comment:
Just a comment, no action required.

It feels like we will be re-creating the list of the `license_required_feature` enum. I already see that cloud storage is different than how it is represented there. Once we are done, we should probably have a look on how to remove the duplication.

Reply:
> It feels like we will be re-creating the list of the `license_required_feature` enum

Good spot. I wish it rendered to a string with naming that's consistent with the docs, though.